### PR TITLE
feat: Sign message

### PR DESCRIPTION
**Pull request by zavodil**
This PR enables the use of the signMessage method from the wallet-selector. The user can sign a message with any of the connected wallets for further [signature verification](https://docs.near.org/build/web3-apps/backend/) on a third-party authentication server.

### CODE REVIEW

**Review by frol — APPROVED** [review]
@evgenykuzyakov Looks good to me, but I don't know if the version update should be part of this PR or a separate effort and whether we should merge it into `dev` branch first

File: src/lib/vm/vm.js [hunk 1]
```diff
@@ -2327,6 +2342,19 @@ export default class VM {
     this.state = {
       ...GlobalInjected,
       ...this.globalFunctions,
+      window: {
+        location: {
+          get hash() {
+            return window.location.hash.substring(1) // skip the first char (#),
```
Comment (evgenykuzyakov) [inline src/lib/vm/vm.js]: Don't need to use the substring here

File: src/lib/vm/vm.js [hunk 2]
```diff
@@ -2327,6 +2342,19 @@ export default class VM {
...
+          },
+          set hash(value) {
+            if (value !== "") {
+              return console.error("Set `window.location.hash` only accepts an empty string as a value.");
```
Comment (evgenykuzyakov) [inline src/lib/vm/vm.js]: Maybe better to throw the error instead of just returning an console.error

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "near-social-vm",
-  "version": "2.6.1",
+  "version": "2.6.2",
```
Comment (evgenykuzyakov) [inline package.json]: It should go to dev branch first

**Review by evgenykuzyakov — COMMENTED** [review]
Add a changelog

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,16 @@
 # Changelog
 
+## 2.6.2
```
Comment (evgenykuzyakov) [inline CHANGELOG.md]:
```suggestion
## Pending
```

File: src/lib/vm/vm.js [hunk 5]
```diff
@@ -1612,6 +1612,21 @@ export default class VM {
     };
 
     const Near = {
+      signMessage: (...args) => {
+        if (args.length < 3) {
```
Comment (evgenykuzyakov) [inline src/lib/vm/vm.js]: Let's add similar API to call, where if the only 1 argument is given and it's an object, then it passes it directly

**Review by evgenykuzyakov — COMMENTED** [review]
Rebased to `dev`

**Review by evgenykuzyakov — APPROVED** [review]

**Pull request merged**